### PR TITLE
Fix `dlite.Instance.from_dict()` to correctly parse ref-type properties

### PR DESCRIPTION
**Pull request by jesper-friis**
# Description
This should make it possible to correctly parse YAML files with ref-type properties.

## Type of change
- [x] Bug fix & code cleanup
- [ ] New feature
- [ ] Documentation update
- [ ] Test update

## Checklist for the reviewer
This checklist should be used as a help for the reviewer.

- [ ] Is the change limited to one issue?
- [ ] Does this PR close the issue?
- [ ] Is the code easy to read and understand?
- [ ] Do all new feature have an accompanying new test?
- [ ] Has the documentation been updated as necessary?

### CODE REVIEW

File: bindings/python/tests/test_ref_type.py [hunk 1]
```diff
@@ -6,10 +6,12 @@
 thisdir = Path(__file__).resolve().parent
 indir = thisdir / "input"
 
-dlite.storage_path.append(indir / "test_ref_type.json")
+dlite.storage_path.append(indir)
+dlite.storage_path.append(indir/"test_ref_type_middle.yaml")
```
Comment (francescalb) [inline bindings/python/tests/test_ref_type.py:10]:
```suggestion
dlite.storage_path.append(indir / "test_ref_type_middle.yaml")
```

File: bindings/python/tests/test_ref_type.py [hunk 2]
```diff
@@ -6,10 +6,12 @@
...
 
+# Note, we read Middle from yaml file, which contains v0.2
 Top = dlite.get_instance("http://onto-ns.com/meta/0.1/Top")
-Middle = dlite.get_instance("http://onto-ns.com/meta/0.1/Middle")
+Middle = dlite.get_instance("http://onto-ns.com/meta/0.2/Middle")
```
Comment (francescalb) [inline bindings/python/tests/test_ref_type.py:14]: I think we should still test json as well

**Review by francescalb — COMMENTED** [review]
I reading the yaml version only available in python?

File: bindings/python/tests/test_ref_type.py (hunk 2 above)
Comment (jesper-friis) [reply bindings/python/tests/test_ref_type.py:14]: We are still testing JSON, since all the other datamodels are read from `input/test_ref_type.json`.

**Comment by jesper-friis** [comment]
> I reading the yaml version only available in python?

Sorry, unsure what you mean here? The YAML plugin is written in Python, but it is still be used from C via the embedded interpreter.

File: bindings/python/tests/test_ref_type.py [hunk 3]
```diff
@@ -1,15 +1,24 @@
...
 indir = thisdir / "input"
 
-dlite.storage_path.append(indir / "test_ref_type.json")
+dlite.storage_path.append(indir)
+dlite.storage_path.append(indir / "test_ref_type_middle.yaml")
+
+# If yaml is available, we read Middle v0.2, which is defined in
+# `test_ref_type_middle.yaml`.  Otherwise, we read Middle v0.1, which
+# is defined together with the other datamodels in `test_ref_type.json`.
+version = "0.2" if yaml else "0.1"
```
Comment (francescalb) [inline bindings/python/tests/test_ref_type.py:19]: In the test suite we have both yaml and json, we should test both.

**Review by francescalb — APPROVED** [review]

**Pull request merged**